components/interop/interop-service/src/main/kotlin/net/corda/interop/service/FacadeToFlowMapperService.kt

```
@@ -0,0 +1,55 @@
+import org.slf4j.LoggerFactory
+
+@Component(service = [FacadeToFlowMapperService::class])
+class FacadeToFlowMapperService @Activate constructor(
+    @Reference(service = VirtualNodeInfoReadService::class)
+    private val virtualNodeInfoReadService: VirtualNodeInfoReadService,
+    @Reference(service = CpiInfoReadService::class)
+    private val cpiInfoReadService: CpiInfoReadService
+) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(this::class.java.enclosingClass)
+        private val mapper = ObjectMapper(YAMLFactory()).registerModule(KotlinModule())
+    }
+    fun getFlowName(
+        destinationIdentity: HoldingIdentity,
+        facadeId: String,
+        facadeName: String
+    ): String {
+        logger.info("Fetching flow name for holding identity : ${destinationIdentity.x500Name} " +
+                "facade id : $facadeId facade name : $facadeName")
+        //Now we are using hardcoded facade definition and
+        // the assumption is we will get the required facade to flow mapping configuration as a part of CPIMetaData.
+        // The following code snippet suggest that how we can read it when it will be available.
+        val vNodeInfo = virtualNodeInfoReadService.get(destinationIdentity)
+        if(vNodeInfo != null) {
+            checkNotNull(vNodeInfo) {
+                "Failed to find the virtual node info for holder '${destinationIdentity}' in ${virtualNodeInfoReadService::class.java}"
+            }
+            val cpiMetadata = cpiInfoReadService.get(vNodeInfo.cpiIdentifier)
+            checkNotNull(cpiMetadata) { "Failed to find the CPI meta data for '${vNodeInfo.cpiIdentifier}}'" }
+            check(cpiMetadata.cpksMetadata.isNotEmpty()) { "No CPKs defined for CPI Meta data id='${cpiMetadata.cpiId}'" }
+            return cpiMetadata.cpksMetadata.first().cordappManifest.attributes.get("") ?: ""
+        } else {
+            return mapper.readValue(
```

Would be better to integrated  the "temporary code" almost as it would the production code - so any code routine shared by temp and not temp code is coded only once ( with very minimal way just fallback to hardcoded one). More specifically there should be single occurrence of `mapper.readValue` which takes value from the first branch  or the hardcoded one if the "prod" returns null:
````
val mappings =  //  try { /*getting from vNodeInfo (the whole if not null branch)*/ } catch (e: Exception) { log.warn() 
 null }?: this::class.java.getResource("/dummy-facade-to-flow-config.yaml")?.readText()
mapper.readValue(mappings, ....)
````

---

Leverage Kotlin expression nature of if else (as I posted in the previous comment) , ignore my formatting in the snippet below:
```
 val content =
        if (vNodeInfo != null) {
            ...
            cpiMetadata.cpksMetadata.first().cordappManifest.attributes[FACADE_TO_FLOW_MAPPING]?:""
        } else {
            this::class.java.getResource("/dummy-facade-to-flow-config.yaml")?.readText().toString()
        }
```